### PR TITLE
[Feature]#137: Add pandas DF -> AnnData and AnnData -> pandas DF

### PR DESCRIPTION
**Pull request by Imipenem**
**PR Checklist**

-   [x] This comment contains a description of changes (with reason)
-   [x] Referenced issue is linked
-   [x] If you've fixed a bug or added code that should be tested, add tests!
-   [x] Documentation in `docs` is updated

**Description of changes**

- implemented a function to convert a given Anndata object to a pandas dataframe

- moved the existing df_to_anndata function from IO part to util part and expose it via the API

### CODE REVIEW

**Comment by Imipenem** [comment]
@Zethson not quite happy with location of the two functions. Any suggestions?

File: ehrapy/api/_anndata_util.py [hunk 1]
```diff
@@ -1,9 +1,100 @@
...
+                "Make sure you spelled the column names correctly."
+            )
+    else:
+        obs = pd.DataFrame(index=df.index.map(str))
+
+    return BaseDataframes(obs, df)
+
+
+def anndata_to_df(adata: AnnData, add_from_obs: list[str] | str | None = None) -> pd.DataFrame:
+    """Transform an AnnData object to a pandas Dataframe
```
Comment (Zethson) [inline ehrapy/api/_anndata_util.py:75]:
```suggestion
    """Transform an AnnData object to a pandas Dataframe.

```

**Review by Zethson — CHANGES_REQUESTED** [review]
LGTM

Three things:
1. Please add a few tests. Think these methods can be well tested.
2. Please add these functions to the documentation
3. Let's move these functions to somewhere else as discussed. Where? TBD.

**Comment by Zethson** [comment]
I think that your submodule is not up to date. You can set git pull to also always pull the submodule. Google how to.

**Review by Zethson — COMMENTED** [review]
I'd suggest that you use a setup_method in the tests to create these objects? Think you have some things that are repeated.

**Comment by Imipenem** [comment]
> I'd suggest that you use a setup_method in the tests to create these objects? Think you have some things that are repeated.


Hm I see your point but unfortunately, this would require a lot of customization. Each test function requires different columns and test values, not sure if its worth the effort.

**Comment by Zethson** [comment]
If it allows you to more quickly move to the other issues I am fine with keeping them ugly for now. Although it hurts my soul ^_^

**Comment by Imipenem** [comment]
> If it allows you to more quickly move to the other issues I am fine with keeping them ugly for now. Although it hurts my soul ^_^

Wrote before

> If it allows you to more quickly move to the other issues I am fine with keeping them ugly for now. Although it hurts my soul ^_^

Think first, write then -> applies to my comment from above^^

**Pull request merged**